zcash_client_sqlite/src/lib.rs

```diff
@@ -85,6 +85,8 @@ use {
     std::{fs, io},
 };
 
+pub type AccountId = u32;
```

Comment:
I'd prefer this to be a "newtype wrapper", i.e. a zero-cost struct wrapper that doesn't expose its internals. Type aliases are useful for making simplifying aliases for more complex types, but they don't provide type safety.

Reply:
There isn't any type safety to be had here. sqlite would accept or produce all u32 values. Making this a struct just means a bunch of code changes to wrap and unwrap the new struct every time the u32 is accessed. Are you sure this is worthwhile?
What if I didn't expose this as a public type at all? But then I guess folks who need to specify a type would just hard-code `u32`, which isn't great.

Reply:
The type alias however allows consumers to use it instead of `u32` so that if we ever _did_ want to replace it with a struct, we could do so without breaking consumer code.

Reply:
I guess an opaque struct now would immediately cause breaks in any code that had been treating it as a zip32 account index, so... that's probably a huge plus in your suggestion.

Reply:
The point of adding an explicit wrapper is that it prevents you from, for example, doing silly things like adding two `AccountId` values together, or using an `AccountId` where you actually meant some other use of `u32`. Even though it's a little bit of extra boilerplate, it's very helpful to have the type system prevent these additional kinds of errors.

Reply:
You'll notice that we do this with a whole bunch of types throughout the codebase: `BlockHeight`, the original `AccountId`, and so forth.

Reply:
Done.

Reply:
Pushed the fix.